controllers/powerconfig_controller.go

```diff
@@ -241,7 +241,8 @@ func (r *PowerConfigReconciler) Reconcile(c context.Context, req ctrl.Request) (
 		}
 
 		powerNode.Spec.CustomDevices = customDevices
-		err := r.Client.Update(context.TODO(), powerNode)
+		patch := client.MergeFrom(powerNode.DeepCopy())
+		err = r.Client.Status().Patch(context.TODO(), powerNode, patch)
```

Comment:
As far as I could understand with the way the Patch call is written we won't be running with optimistic locking  which means we'd be overriding the customDevices field without trying to reconcile what's in etcd. 
I'm a bit torn whether this particular call is the right thing to do. 
Powernodes don't seem the powerconfig's main responsibility so I don't think it should be updating the customDevices field like that. I'd prefer to get into conflicts and trigger as many reconciler loops as needed to solve the conflict.
Now, on the other side, this is the only place (for the moment) where we update the powernode's customDevices field so there are no issues of concurrent updating.